torch/csrc/distributed/rpc/tensorpipe_agent.cpp

```diff
@@ -354,11 +354,38 @@ void TensorPipeAgent::prepareNames() {
   }
 }
 
+void TensorPipeAgent::checkAndSetStaticGroup(
+    const c10::intrusive_ptr<::c10d::Store>& store) {
+  std::string isStaticGroupKey("rpcIsStaticGroup");
+  bool isStaticGroupKeyExists =
+      store->check(std::vector<std::string>{isStaticGroupKey});
+  if (isStaticGroupKeyExists) {
+    std::vector<uint8_t> isStaticGroupVector = store->get(isStaticGroupKey);
+    bool storeIsStaticGroup = false;
+    std::istringstream(std::string(
+        (char*)isStaticGroupVector.data(), isStaticGroupVector.size())) >>
+        std::boolalpha >> storeIsStaticGroup;
+    TORCH_CHECK(
+        isStaticGroup_ == storeIsStaticGroup,
+        fmt::format(
+            "RPC group behavior is different than expected. isStaticGroup_ is initialized with {} while store is {}",
+            isStaticGroup_,
+            storeIsStaticGroup))
+  } else {
+    std::string isStaticGroupStr;
+    isStaticGroupStr = isStaticGroup_ ? "true" : "false";
+    std::vector<uint8_t> isStaticGroupVec(
+        (uint8_t*)isStaticGroupStr.c_str(),
+        (uint8_t*)isStaticGroupStr.c_str() + isStaticGroupStr.length());
+    store->set(isStaticGroupKey, isStaticGroupVec);
```

Comment:
Correct compareSet does not have a good way of verifying whether the value was set or previously existed. Using the regular set() makes the logic simpler.

Reply:
Thanks, there is a race condition because this is used in static TP initialization. Curious, what did you envision to be the purpose of this method? I assumed this was to do a sanity check to make sure that there are not both static and dynamic RPC groups being initialized. If so, I think I can use compareSet() to fix the race condition.

Reply:
>  Curious, what did you envision to be the purpose of this method? I assumed this was to do a sanity check to make sure that there are not both static and dynamic RPC groups being initialized.

Yep, exactly